src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/VideoLinkBookingSpecification.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.digital.hmpps.whereabouts.dto
+
+import io.swagger.annotations.ApiModel
+import io.swagger.annotations.ApiModelProperty
+import java.time.LocalDateTime
+import javax.validation.Valid
+import javax.validation.constraints.NotEmpty
+import javax.validation.constraints.NotNull
+
+@ApiModel(description = " Video Link Booking details")
+data class VideoLinkBookingSpecification(
+
+  @ApiModelProperty(value = "Offender booking Id", example = "1")
+  @field:NotNull
+  val bookingId: Long?,
+
+  @ApiModelProperty(
+    value = "The location of the court that requires the appointment",
+    example = "York Crown Court",
+    required = true
+  )
+  @NotEmpty
+  val court: String,
+
+  @ApiModelProperty(value = "Booking placed by the court", required = false)
+  val madeByTheCourt: Boolean = true,
+
+  @ApiModelProperty(value = "Free text comments", example = "Requires special access")
+  val comment: String? = null,
+
+  @ApiModelProperty(value = "Pre-hearing appointment", required = false)
+  @Valid
+  val pre: VideoLinkAppointmentSpecification? = null,
+
+  @ApiModelProperty(value = "Main appointment", required = true)
+  @field:Valid
+  val main: VideoLinkAppointmentSpecification,
+
+  @ApiModelProperty(value = "Post-hearing appointment", required = false)
+  @Valid
+  val post: VideoLinkAppointmentSpecification? = null
+)
+
+@ApiModel(description = "Detail of a single Video Link Appointment, either pre, main or post")
+data class VideoLinkAppointmentSpecification(
+
+  @ApiModelProperty(value = "The identifier of the appointment's location", example = "1", required = true)
+  @field:NotNull
+  val locationId: Long?,
```

Comment:
Is this nullable by mistake?

Reply:
It is deliberate.  Making it nullable enables validation.  It seems that non-nullable fields are always populated even if there's no value in the request.  In this case if locationId were {{val locationId: Long,}} then Spring would use a default value of 0.  I did think about creating a 'NotZero' validator, got as far as writing it, but decided that approach was just ugly and wrong.